content/en/docs/languages/python/propagation.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Propagation
+description: Context propagation for the Python SDK
+aliases: [api/propagation]
+weight: 65
+cSpell:ignore: tracestate
+---
+
+Propagation is the mechanism that moves data between services and processes.
+Although not limited to tracing, it is what allows traces to build causal
+information about a system across services that are arbitrarily distributed
+across process and network boundaries.
+
+## Manual W3C Trace Context Propagation
+
+The following generic example demonstrates how you can propagate trace context
+manually.
+
+First, on the sending service, you'll need to inject the current `context`:
+
+```python
+from flask import Flask
+import requests
+from opentelemetry import trace, propagators, baggage
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+from opentelemetry.baggage.propagation import W3CBaggagePropagator
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter, BatchSpanProcessor
+
+app = Flask(__name__)
+
+trace.set_tracer_provider(TracerProvider())
+trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(ConsoleSpanExporter()))
+
+tracer = trace.get_tracer(__name__)
+
+@app.route('/')
+def hello():
+    with tracer.start_as_current_span("api1_span") as span:
+        ctx = baggage.set_baggage("hello", "world")
+
+        headers = {}
+        W3CBaggagePropagator().inject(headers, ctx)
+        TraceContextTextMapPropagator().inject(headers, ctx)
+        print(headers)
+
+        response = requests.get('http://127.0.0.1:5001/', headers=headers)
+        return f"Hello from API 1! Response from API 2: {response.text}"
+
+if __name__ == '__main__':
+    app.run(port=5002)
+```
+
+On the receiving service, you'll need to extract `context` (for example, from
+parsed HTTP headers) and then set them as the current trace context.
+
+```python
+from flask import Flask, request
+from opentelemetry import trace, baggage
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter, BatchSpanProcessor
+from opentelemetry.baggage.propagation import W3CBaggagePropagator
+
+app = Flask(__name__)
+
+trace.set_tracer_provider(TracerProvider())
+trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(ConsoleSpanExporter()))
+
+tracer = trace.get_tracer(__name__)
+
+@app.route('/')
+def hello():
+    # Example: Log headers received in the request in API 2
+    headers = dict(request.headers)
+    print(f"Received headers: {headers}")
+    carrier ={'traceparent': headers['Traceparent']}
```

Comment:
headers are case insensitive so passing the headers directly as carrier should work

Reply:
![image](https://github.com/open-telemetry/opentelemetry.io/assets/13224827/7da5f6d2-7ceb-4981-9991-61619375135e)
it seems like directly passing headers not working on my local testing. Should I try another Propagator?